### DISCUSSION TITLE
Is there a method to persist metrics values to file or any other custom storage?

### DISCUSSION THREAD

**Discussion by navrocky** [open-telemetry/opentelemetry-cpp#2996] [Q&A]
My problem is that my service periodically restarts and the metrics are reset. Is there some standard way to save them to a file or to my own custom storage to restore them after restart?

**Comment by lalitb** [comment]
Which backend are you using? If it is Prometheus, it knows how to handle the reset scenario.

**Reply by navrocky** [reply]
Yes, I'm using prometheus. Do I have to do something extra to make it save metrics or does it already work out of the box?

**Reply by lalitb** [reply]
You won’t do anything in the Otel SDK side. Prometheus provides query functions like rate/irate to handle reset scenario. In case this thread helps - https://www.reddit.com/r/PrometheusMonitoring/s/rZyiU3VyM1
